distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Widget_Helpers.enso

```diff
@@ -156,18 +162,23 @@ make_join_kind_selector display=Display.Always =
 
 ## PRIVATE
    Make a join condition selector.
-make_join_condition_selector : Table -> Display -> Widget
-make_join_condition_selector table display=Display.Always =
-    col_names_selector = make_column_name_selector table display=Display.Always
+make_join_condition_selector : Table -> Display -> Function -> Widget
+make_join_condition_selector table display=Display.Always cache=Nothing =
+    right = if cache.is_nothing.not then cache "right"
+
+    col_names = intersect_column_names table right
+    self_selector = make_column_name_selector table display=Display.Always
+    right_selector = if right.is_nothing then Text_Input else
+        make_column_name_selector right display=Display.Always
 
     fqn = Meta.get_qualified_type_name Join_Condition
-    equals = Option "Equals" fqn+".Equals" [["left", col_names_selector], ["right", Text_Input]]
-    equals_ci = Option "Equals (Ignore Case)" fqn+".Equals_Ignore_Case" [["left", col_names_selector], ["right", Text_Input]]
-    between = Option "Between" fqn+".Between" [["left", col_names_selector], ["right_lower", Text_Input], ["right_upper", Text_Input]]
+    equals = Option "Equals" fqn+".Equals" [["left", self_selector], ["right", right_selector]]
+    equals_ci = Option "Equals (Ignore Case)" fqn+".Equals_Ignore_Case" [["left", self_selector], ["right", right_selector]]
+    between = Option "Between" fqn+".Between" [["left", self_selector], ["right_lower", right_selector], ["right_upper", right_selector]]
```

Comment:
As @AdRiley pointed out, there have to be [separate self and right selectors](https://github.com/enso-org/enso/pull/9410/commits/19e8d5b2def58f00b50dc1555c91bb7414fc610d) - with that we see two different dropdowns for tables with `a b e` columns and `c b e` column:

![obrazek](https://github.com/enso-org/enso/assets/26887752/dbc6ca2e-d422-4cdf-9232-58c88d199ced)
![obrazek](https://github.com/enso-org/enso/assets/26887752/d83d6e25-abce-499f-82b2-15c65779fa79)